### PR TITLE
Include metrics testing in E2E tests

### PR DESCRIPTION
**Pull request by harshad16**
Include metrics testing in E2E tests

## Description

Included metrics testing check as a part of E2E tests, as mentioned in Note here:https://github.com/opendatahub-io/data-science-pipelines-operator/pull/23#issue-1593637302
- Adjusted Token creation for SA as per cluster rules.

## How Has This Been Tested?
with the help of makefile and openshift cluster, try the following:
```
cd tests
make build
make run
```
Tested on openshift cluster with version: 4.12.2

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: tests/basictests/dsp-operator.sh [hunk 1]
```diff
@@ -49,7 +67,20 @@ function verify_pipeline_availabilty() {
 
 function create_run() {
     header "Creating the run from uploaded pipeline"
-    RUN_ID=$(curl -s -k -H "Authorization: Bearer ${SA_TOKEN}" -H "Content-Type: application/json"  -d "{\"name\":\"test-pipeline-run_run\", \"pipeline_spec\":{\"pipeline_id\":\"${PIPELINE_ID}\"}}" https://${ROUTE}/apis/v1beta1/runs | jq -r .run.id)
+
+    RUN_ID=$(( curl -k -H "Authorization: Bearer ${SA_TOKEN}" \
+        -H "Content-Type: application/json" \
+        -X POST "https://${ROUTE}/apis/v1beta1/runs" \
+        -d @- << EOF
+        {
```
Comment (DharmitD) [inline tests/basictests/dsp-operator.sh:75]: Nice job spacing out the json here!

**Review by DharmitD — COMMENTED** [review]
/lgtm

**Review by DharmitD — COMMENTED** [review]
/lgtm

File: tests/resources/dsp-operator/test-dspo-cr.yaml [hunk 2]
```diff
@@ -19,36 +19,13 @@ spec:
     image: quay.io/modh/odh-ml-pipelines-viewercontroller-container:v1.18.0-8
   mlpipelineUI:
     image: quay.io/opendatahub/odh-ml-pipelines-frontend-container:beta-ui
-    configMap: ds-pipeline-ui-configmap
+    configMap: ds-pipeline-ui-configmap-sample
```
Comment (HumairAK) [inline tests/resources/dsp-operator/test-dspo-cr.yaml:22]: don't need this field, that's basically the default configmap

Comment (harshad16) [reply tests/resources/dsp-operator/test-dspo-cr.yaml]: Made the required changes.
Ready for review again @HumairAK 
thanks :smile:

**Pull request merged**